r/tests/testthat/test-expression.R

```diff
@@ -42,8 +69,19 @@ test_that("Expression task works with one gene", {
   req$body$genes <- list("MS4A1")
 
   res <- runExpression(req, data)
-  expect_equal(ncol(res$rawExpression), 1)
-  expect_equal(nrow(res$rawExpression), ncol(data))
+  expect_equal(length(res), 1)
+  expect_equal(length(res[[1]]$rawExpression$expression), ncol(data))
+})
+
+test_that("If the max truncated expression value is 0, iterates correctly to find a non zero value", {
+  data <- mock_scdata()
+  req <- mock_req()
+
+  data@assays$RNA@data["MS4A1",] <- 0
+  data@assays$RNA@data["MS4A1",1] <- 5
+
+  res <- runExpression(req, data)
+  expect_false(all(res$MS4A1$truncatedExpression$expression==0) )
 })
 
 test_that("runExpression throws an error if request only non existing genes", {
```

Comment:
what happens if the user searches for, say 3 genes: one that doesn't exist and two that exist? Will the code still throw an error and not return all of them? Or will it return only the genes that it have?

Reply:
Addressed in a new ticket!